src/lib/transport/ip/tcp_connect.c

```diff
@@ -295,8 +295,11 @@ oo_sp ci_tcp_connect_find_local_peer(ci_netif *ni, int locked,
 #if CI_CFG_IPV6
 static void ci_tcp_init_ipcache_ip4_hdr(ci_tcp_state* ts)
 {
+  ci_uintptr_t tcphdr_offs_new = (ci_uintptr_t)TS_TCP(ts);
+  ci_uintptr_t tcphdr_offs_old = (ci_uintptr_t)TS_IP6_TCP(ts);
```

Comment:
why not type it as `void *` instead of `uintptr`?

Reply:
The compiler still generates the warning if I use `void*`. Only cast to `int` helps.